src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -266,6 +270,8 @@ void
 mongoc_client_encryption_opts_set_keyvault_client (
    mongoc_client_encryption_opts_t *opts, mongoc_client_t *keyvault_client)
 {
+   BSON_ASSERT_PARAM (keyvault_client);
+
```

Comment:
```suggestion
```

Suggest removing this assertion. A caller may be relying on passing `NULL` to clear the option.